src/digraph.rs

```diff
@@ -2720,6 +2720,45 @@ impl PyDiGraph {
         self.clone()
     }
 
+    /// Reverse the direction of all edges in the graph, in place.
+    ///
+    /// This method modifies the graph instance to reverse the direction of all edges.
+    /// It does so by iterating over all edges in the graph and removing each edge,
+    /// then adding a new edge in the opposite direction with the same weight.
+    ///
+    ///  .. note:: The edge indices will change if this method is used to reverse the edges
+    ///
+    /// For example::
+    ///
+    ///     import rustworkx as rx
+    ///
+    ///     graph = rx.PyDiGraph()
+    ///     graph.add_nodes_from(range(4))
+    ///     graph.add_edge(0, 1, 3)
+    ///     graph.add_edge(1, 2, 5)
+    ///     graph.add_edge(2, 3, 2)
+    ///
+    ///     graph.reverse_inplace()
+    ///
+    ///     assert graph.edge_list() == [(3, 2), (2, 1), (1, 0)]
+    #[pyo3(text_signature = "(self)")]
+    pub fn reverse_inplace(&mut self, py: Python) {
```

Comment:
Because we do not have a reverse method, you can call it reverse

Reply:
Good Idea, I thought that because it behaves differently to not name it the same way (the copy option which I found hard to implement easily) 
if you want I can add it (maybe in a separate PR)